### PR TITLE
add React Hooks rules

### PR DESCRIPTION
**Pull request by willopez**
Adds new rules for React Hooks, see https://www.npmjs.com/package/eslint-plugin-react-hooks for details.

### CODE REVIEW

File: index.json [hunk 1]
```diff
@@ -550,6 +551,8 @@
     "react/self-closing-comp": "error",
     "react/sort-comp": "error",
     "react/sort-prop-types": "error",
+    "react-hooks/rules-of-hooks": "error",
+    "react-hooks/exhaustive-deps": "warn",
```
Comment (kieckhafer) [inline index.json]: @willopez Let's make this an `error` here, and then if we need to we can override all the new rules with `warn` inside of `reaction`, that way they aren't blockers right off the bat while we are getting used to them / seeing exactly how they function.

https://github.com/reactioncommerce/reaction/blob/master/package.json#L396

Comment (willopez) [reply index.json]: Sounds good, updated.

**Comment by kieckhafer** [comment]
@willopez please fix the DCO check, you can merge once that is fixed.

**Comment by willopez** [comment]
@kieckhafer Signed off and added `feat` to the last commit.

**Pull request merged**